File: network-tasks/src/test/java/io/appmetrica/analytics/networktasks/impl/HashCodeEqualsTest.java
package io.appmetrica.analytics.networktasks.impl;

import io.appmetrica.analytics.networktasks.internal.RetryPolicyConfig;
import java.util.Arrays;
import java.util.Collection;
import nl.jqno.equalsverifier.EqualsVerifier;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.ParameterizedRobolectricTestRunner;

public class HashCodeEqualsTest {
    @RunWith(ParameterizedRobolectricTestRunner.class)
    public static class AllFieldsTest {

        private final Class mClazz;

        public AllFieldsTest(final Class clazz) {
            mClazz = clazz;
        }

        @ParameterizedRobolectricTestRunner.Parameters(name = "class = {0}")
        public static Collection<Object[]> params() {
            return Arrays.asList(new Object[][]{
                    {RetryPolicyConfig.class}
            });
        }

        @Test
        public void testEqualsAndHashCode() {
            EqualsVerifier.forClass(mClazz)
                    .usingGetClass()
                    .verify();
        }
    }
}

File: id-sync/src/test/java/io/appmetrica/analytics/idsync/HashCodeEqualsTest.kt
package io.appmetrica.analytics.idsync

import io.appmetrica.analytics.idsync.internal.model.IdSyncConfig
import io.appmetrica.analytics.idsync.internal.model.Preconditions
import io.appmetrica.analytics.idsync.internal.model.RequestConfig
import io.appmetrica.analytics.testutils.CommonTest
import nl.jqno.equalsverifier.EqualsVerifier
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.ParameterizedRobolectricTestRunner

@RunWith(ParameterizedRobolectricTestRunner::class)
class HashCodeEqualsTest(private val clazz: Class<Any?>) : CommonTest() {

    companion object {

        @ParameterizedRobolectricTestRunner.Parameters(name = "#{index} - {0}")
        @JvmStatic
        fun data(): Collection<Array<Any?>> = listOf(
            arrayOf(IdSyncConfig::class.java),
            arrayOf(Preconditions::class.java),
            arrayOf(RequestConfig::class.java)
        )
    }

    @Test
    fun hashCodeAndEquals() {
        EqualsVerifier.forClass<Any?>(clazz)
            .usingGetClass()
            .verify()
    }
}
